src/dswx_sar/save_mgrs_tiles.py

```diff
@@ -1,73 +1,114 @@
 
-def get_geographic_boundaries_from_mgrs_tile(mgrs_tile_name, verbose=False):
-    """Get bounding box cooridnate from MGRS tile
+def get_geographic_boundaries_from_mgrs_tile(mgrs_tile_name,
+                                             mgrs_db_path=None):
+    """Get geographic extents from given MGRS tile
     Parameters
     ----------
     mgrs_tile_name: str
-        MGRS tile grid ID (e.g. 10SEG)
+        Name of MGRS tile (ex. 18LWQ)
+    mgrs_db_path : str
+        path for MGRS database
+
     Returns
     -------
-    x_value_min: float
+    minx: float
         minimum x cooridate (UTM) for given MGRS tile
-    x_value_max: float
+    maxx: float
         maximum x cooridate (UTM) for given MGRS tile
-    y_value_min: float
+    miny: float
         minimum y cooridate (UTM) for given MGRS tile
-    y_value_max: float
+    maxy: float
         maximum y cooridate (UTM) for given MGRS tile
+    epsg: int
+        epsg code
     """
-    mgrs_obj = mgrs.MGRS()
-    lower_left_utm_coordinate = mgrs_obj.MGRSToUTM(mgrs_tile_name)
-    utm_zone = lower_left_utm_coordinate[0]
-    is_northern = lower_left_utm_coordinate[1] == 'N'
-    x_min = lower_left_utm_coordinate[2]
-    y_min = lower_left_utm_coordinate[3]
-
-    # compute boundaries
-    x_list = []
-    y_list = []
-    for offset_x_multiplier in range(2):
-        for offset_y_multiplier in range(2):
-            x = x_min + offset_x_multiplier * 109.8 * 1000
-            y = y_min + offset_y_multiplier * 109.8 * 1000
-            x_list.append(x)
-            y_list.append(y)
-
-    x_value_min = np.min(x_list)
-    x_value_max = np.max(x_list)
-    y_value_min = np.min(y_list)
-    y_value_max = np.max(y_list)
-
-    return x_value_min, x_value_max, y_value_min, y_value_max
-
-def save_mgrs_tile(source_tif_path,
-                   output_dir_path,
-                   output_tif_name,
-                   output_mgrs_id,
-                   method='nearest',
-                   no_data_value=255):
+    # if mgrs tile database is not given,
+    # return the coordinates based on python mgrs
+    if mgrs_db_path is None:
+        mgrs_obj = mgrs.MGRS()
+        lower_left_utm_coordinate = mgrs_obj.MGRSToUTM(mgrs_tile_name)
+
+        x_min = lower_left_utm_coordinate[2]
+        y_min = lower_left_utm_coordinate[3]
+
+        if lower_left_utm_coordinate[1] == 'N':
+            is_southern = False
+        else:
+            is_southern = True
+
+        crs = CRS.from_dict({'proj': 'utm',
+                             'zone': lower_left_utm_coordinate[0],
+                             'south': is_southern})
+
+        epsg = crs.to_authority()[1]
+
+        # compute boundaries
+        x_list = []
+        y_list = []
+        for offset_x_multiplier in range(2):
+            for offset_y_multiplier in range(2):
+                # additional coordinates to compute min/max of x/y
+                x_more = x_min + offset_x_multiplier * 109.8 * 1000
```

Comment:
You also need to add the margin to the `x_min` and `y_min` coordinates:
https://github.com/nasa/PROTEUS/blob/e5b411a91d5e78bc1479ba6d146ce62a29823e49/src/proteus/core.py#L126C18-L126C67